dozer/cogs/shortcuts.py

```diff
@@ -129,29 +130,52 @@ async def list(self, ctx):
     """
 
     @Cog.listener()
-    async def on_message(self, msg):
+    async def on_message(self, msg: discord.Message):
         """prefix scanner"""
         if not msg.guild or msg.author.bot:
             return
-        setting = await self.settings_cache.query_one(guild_id=msg.guild.id)
-        if setting is None:
-            return
-
-        c = msg.content
-        if len(c) < len(setting.prefix):
-            return
 
-        if not c.startswith(setting.prefix):
+        setting = await ShortcutSetting.get_unique_by(guild_id = msg.guild.id)
+        if not setting:
             return
 
-        shortcuts = await ShortcutEntry.get_by(guild_id=msg.guild.id)
-        if not shortcuts:
-            return
-
-        for shortcut in shortcuts:
-            if c.lower()[len(setting.prefix):] == shortcut.name.lower():
-                await msg.channel.send(shortcut.value)
-                return
+        # Search for the prefix within the message content
+        prefix = setting.prefix
+        prefix_index = msg.content.find(prefix)
+
+        # Check for the presence of the prefix anywhere in the message
+        prefix_index = msg.content.find(prefix)
+
+        if prefix_index != -1:
```

Comment:
Since we only check one prefix index, a message like
> Hi! Check out !gm1.
would not work properly. We should iterate over all possible (non-overlapping) matches. This also would allow the use of multiple shortcuts in one message.

Reply:
Are you sure? What if someone has like 5 ! or something in their message? Or, if they just want to spam like 50 ! ?

Reply:
:woman_shrugging: it’d be a weird edge case to not work. If someone put 50 exclamation marks, it will lag things a bit but probably not be too bad? Maybe add a cap at like 10 prefixes checked or something.